lib/improver/nowcasting/nowcasting.py

```diff
@@ -0,0 +1,196 @@
+
+        if operation in possible_operations:
+            self.operation = operation
+        else:
+            msg = ("Operation '{}' not supported for combining "
+                   "precipitation rate and "
+                   "orographic enhancement.".format(operation))
+            raise ValueError(msg)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ApplyOrographicEnhancement: operation: {}>')
+        return result.format(self.operation)
+
+    @staticmethod
+    def _extract_orographic_enhancement_cube(precip_cube, oe_cubes):
+        """Extract the orographic enhancement cube with the required time
+        coordinate.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation fields.
+            oe_cubes (iris.cube.Cube or iris.cube.CubeList):
+                Cube or CubeList containing the orographic enhancement fields.
+
+        Returns:
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement fields at the
+                required time.
+
+        """
+        time_point, = iris_time_to_datetime(precip_cube.coord("time"))
+        oe_cube = extract_nearest_time_point(oe_cubes, time_point)
+        return oe_cube
+
+    def _apply_cube_combiner(self, precip_cube, oe_cube):
+        """Combine the precipitation rate cube and the orographic enhancement
+        cube.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation field.
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement field.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field modified by the
+                orographic enhancement cube.
+
+        """
+        # Ensure the orographic enhancement cube matches the
+        # dimensions of the precip_cube.
+        oe_cube = check_cube_coordinates(precip_cube, oe_cube)
+
+        # Ensure that orographic enhancement is in the units of the
+        # precipitation rate cube.
+        oe_cube.convert_units(precip_cube.units)
+
+        # Set orographic enhancement to be zero for points with a
+        # precipitation rate of < 1/32 mm/hr.
+        original_units = Unit("mm/hr")
+        threshold_in_cube_units = (
+            original_units.convert(1/32., precip_cube.units))
+        oe_cube.data[precip_cube.data < threshold_in_cube_units] = 0.
+
+        # Use CubeCombiner to combine the cubes.
+        temp_cubelist = iris.cube.CubeList([precip_cube, oe_cube])
+        cube = CubeCombiner(self.operation).process(
+            temp_cubelist, precip_cube.name())
+        return cube
+
+    @staticmethod
+    def apply_minimum_precip_rate(cube, tolerance=-1e-9):
+        """Ensure that negative precipitation rates are capped at +1/32 mm/hr.
+
```

Comment:
I'm not sure this function does quite the right thing.  I think we want any number less than a minimum rate (ie 1/32 mm/h) **where there was precipitation in the input field** to be "capped" at a minimum of 1/32.   I think perhaps a better way of doing this would be:

- When the operation is "subtract", create a "precip > 1/32 mm/h" mask
- Run the operation
- Apply the minimum precipitation rate of 1/32 mm/h only for the "subtract" operation, where the mask is valid

I think again we want this minimum rate to be a class member, not a "tolerance".

Reply:
I've modified this function, hopefully this captures the intended usage.